### PR TITLE
feat(dlp): added a sample for inspect string without overlap

### PR DESCRIPTION
**Pull request by AarshD-crest**
- Added a sample for [inspect string without overlap](https://cloud.google.com/dlp/docs/samples/dlp-inspect-string-without-overlap) with unit test-case.

### CODE REVIEW

File: dlp/snippets/inspect/inspect_string_without_overlap.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Closing the client safely cleans up background resources.
+	defer client.Close()
+
+	// Specify the type and content to be inspected.
+	var contentItem = &dlppb.ContentItem{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_without_overlap.go:47]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_string_without_overlap.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+			ByteItem: &dlppb.ByteContentItem{
+				Type: dlppb.ByteContentItem_TEXT_UTF8,
+				Data: []byte(textToInspect),
+			},
+		},
+	}
+
+	// Specify the type of info the inspection will look for.
+	// See https://cloud.google.com/dlp/docs/infotypes-reference for complete list of info types.
+	var infoTypes = []*dlppb.InfoType{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_without_overlap.go:58]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_string_without_overlap.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+		{Name: "DOMAIN_NAME"},
+		{Name: "EMAIL_ADDRESS"},
+	}
+
+	// Define a custom info type to exclude email addresses
+	var customInfotype = &dlppb.CustomInfoType{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_without_overlap.go:64]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_string_without_overlap.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+		InfoType: &dlppb.InfoType{
+			Name: "EMAIL_ADDRESS",
+		},
+		ExclusionType: dlppb.CustomInfoType_EXCLUSION_TYPE_EXCLUDE,
+	}
+
+	// Exclude EMAIL_ADDRESS matches
+	var exclusionRule = &dlppb.ExclusionRule{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_without_overlap.go:72]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_string_without_overlap.go [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+				},
+			},
+		},
+		MatchingType: dlppb.MatchingType_MATCHING_TYPE_PARTIAL_MATCH,
+	}
+
+	// Construct a ruleSet that applies the exclusion rule to the DOMAIN_NAME infoType.
+	// If a DOMAIN_NAME match is part of an EMAIL_ADDRESS match, the DOMAIN_NAME match will
+	// be excluded.
+	var ruleSet = &dlppb.InspectionRuleSet{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_without_overlap.go:86]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_string_without_overlap.go [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+			RuleSet: []*dlppb.InspectionRuleSet{
+				ruleSet,
+			},
+		},
+	}
+
+	// Send the request.
+	resp, err := client.InspectContent(ctx, req)
+	if err != nil {
+		fmt.Fprintf(w, "Receive: %v", err)
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_without_overlap.go:119]: issue: just return the error as-is to caller.

File: dlp/snippets/inspect/inspect_string_without_overlap.go (hunk 1 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_string_without_overlap.go]: fixed it.

File: dlp/snippets/inspect/inspect_string_without_overlap.go (hunk 3 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_string_without_overlap.go]: fixed it.

File: dlp/snippets/inspect/inspect_string_without_overlap.go (hunk 5 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_string_without_overlap.go]: fixed it.

File: dlp/snippets/inspect/inspect_string_without_overlap.go (hunk 2 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_string_without_overlap.go]: fixed it.

File: dlp/snippets/inspect/inspect_string_without_overlap.go (hunk 4 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_string_without_overlap.go]: fixed it.

**Review by annadata — APPROVED** [review]
LGTM

**Review by telpirion — APPROVED** [review]

**Pull request merged**